File: mobileV2/app/src/main/java/com/krishimitra/mobilev2/LanguageFragment.java
package com.krishimitra.mobilev2;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.navigation.fragment.NavHostFragment;

import com.krishimitra.mobilev2.data.SessionManager;
import com.krishimitra.mobilev2.databinding.FragmentLanguageBinding;

public class LanguageFragment extends Fragment {

    private FragmentLanguageBinding binding;
    private SessionManager sessionManager;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = FragmentLanguageBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        sessionManager = new SessionManager(requireContext());

        // Check if language is already selected and user is logged in
        if (sessionManager.getAuthToken() != null) {
            NavHostFragment.findNavController(LanguageFragment.this)
                    .navigate(R.id.action_LanguageFragment_to_HomeFragment);
            return;
        }

        binding.btnContinue.setOnClickListener(v -> {
            String selectedLang = "en";
            if (binding.rbMarathi.isChecked()) selectedLang = "mr";
            else if (binding.rbHindi.isChecked()) selectedLang = "hi";
            
            selectLanguage(selectedLang);
        });
    }

    private void selectLanguage(String lang) {
        sessionManager.saveLanguage(lang);
        NavHostFragment.findNavController(LanguageFragment.this)
                .navigate(R.id.action_LanguageFragment_to_LoginFragment);
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: mobileV2/app/src/main/java/com/krishimitra/mobilev2/LanguageFragment.kt
package com.krishimitra.mobilev2

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.compose.material3.MaterialTheme
import androidx.compose.ui.platform.ComposeView
import androidx.compose.ui.platform.ViewCompositionStrategy
import androidx.fragment.app.Fragment
import androidx.navigation.fragment.findNavController
import com.krishimitra.mobilev2.data.SessionManager

class LanguageFragment : Fragment() {

    private lateinit var sessionManager: SessionManager

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        sessionManager = SessionManager(requireContext())

        // Check if language is already selected and user is logged in
        if (sessionManager.getAuthToken() != null) {
            findNavController().navigate(R.id.action_LanguageFragment_to_HomeFragment)
        }

        return ComposeView(requireContext()).apply {
            setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
            setContent {
                MaterialTheme {
                    LanguageScreen { selectedLang ->
                        selectLanguage(selectedLang)
                    }
                }
            }
        }
    }

    private fun selectLanguage(lang: String) {
        sessionManager.saveLanguage(lang)
        findNavController().navigate(R.id.action_LanguageFragment_to_LoginFragment)
    }
}
